packages/workbox-routing/lib/RegExpRoute.mjs

```diff
@@ -0,0 +1,119 @@
+/*
+ Copyright 2017 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import core from 'workbox-core';
+import {_private} from 'workbox-core';
+
+import '../_version.mjs';
+import Route from './Route.mjs';
+
+/**
+ * RegExpRoute is a helper class to make defining regular expression based
+ * [Routes]{@link module:workbox-routing.Route} easy.
+ *
+ * The matching for regular expressions is slightly different depending on
+ * whether the request URL is
+ * [same-origin](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy),
+ * or cross-origin. Here's why:
+ *
+ * It's common to use a broad pattern like `new RegExp('/styles/')` to match
+ * all URLs served from a `/styles/` directory on your site.
+ *
+ * If we used this in a service worker deployed on `https://example.com`,
+ * this `RegExp` would trigger a match for
+ * <code>https://example.com<strong>/styles/main.css</strong></code>, as
+ * intended.
+ *
+ * However, it's unlikely that we'd intend for this to trigger matches for
+ * <code>https://third-party-origin.com<strong>/styles/example.css</strong></code>.
+ *
+ * To overcome this common issue, `RegExpRoute` will only trigger a match for a
+ * cross-origin request if the `RegExp` provided matches at the *start* of the
+ * cross-origin URL. For same-origin requests, the match can be
+ * *anywhere* within the URL. In both cases, the `RegExp` only needs to match
+ * some portion of the URL, not the full URL.
+ *
+ * If you *do* want a `RegExpRoute` to trigger matches for some cross-origin
+ * requests, then going back to our styles example, you could use
+ * `new RegExp('https://third-party-origin\.com/styles/')`, which will trigger
+ * a match for URLs like
+ * <code><strong>https://third-party-origin.com/styles/example.css</strong></code>.
+ *
+ * If you want to use a single `RegExpRoute` that triggers a match for both
+ * same- and cross-origin requests, then use a `RegExp` that includes a wildcard
+ * that will match the start of any URL, like new `RegExp('.+/styles/')`.
+ *
+ * @memberof module:workbox-routing
+ * @extends Route
+ */
+class RegExpRoute extends Route {
+  /**
+   * Constructor for `RegExpRoute`.
```

Comment:
Can we just drop this description? Templates for JSDoc should highlight it's a constructor.